api/woeip/apps/air_quality/views.py

```diff
@@ -2,40 +2,99 @@
 
 from django.contrib import messages
 from django.contrib.auth.mixins import LoginRequiredMixin
-from django.shortcuts import redirect, render
+from django.shortcuts import redirect, render, get_object_or_404
 from django.views import View
+from rest_framework import generics, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
 
-from .forms import CollectionForm
-from .models import Collection
+from woeip.apps.air_quality.forms import CollectionForm
+from woeip.apps.air_quality import models, serializers
 
 logger = logging.getLogger(__name__)
 
 
-class Upload(LoginRequiredMixin, View):
-    def get(self, request):
-        """Present file upload form to user"""
-        return render(self.request, 'air_quality/upload.html', {
-            'form': CollectionForm
-        })
-
-    def post(self, request):
-        """Save files to SessionData table"""
-        files = self.request.FILES
-        form = CollectionForm(self.request.POST, files)
-        if form.is_valid():
-            form.save()
-            return redirect('view')
-        else:
-            messages.add_message(self.request, messages.ERROR, 'File upload error')
-            return render(self.request, 'air_quality/upload.html', {
-                'form': CollectionForm
-            })
-
-
-class ViewCollection(View):
-    """Provide temporary development page to view all uploaded SessionDatas."""
-    def get(self, request):
-        collection_list = Collection.objects.all()
-        return render(self.request, 'air_quality/view.html', {
-            'collection_list': collection_list
-        })
+class CalibrationViewSet(viewsets.ModelViewSet):
+    queryset = models.Calibration.objects.all()
+    serializer_class = serializers.CalibrationSerializer
+
+    def get_queryset(self):
+        queryset = models.Calibration.objects.all()
+
+        return queryset
+
+
+class CollectionViewSet(viewsets.ModelViewSet):
+    queryset = models.Collection.objects.all()
+    serializer_class = serializers.CollectionSerializer
+
+    @action(detail=True, methods=["GET"])
+    def geo(self, request, pk=None):
```

Comment:
I would suggest something like `data` instead of `geo` since this is getting all the data (not just geotags).

Reply:
Yeah I'm good with that!